network/validator/pubsub/authorized_sender_validator_test.go

```diff
@@ -0,0 +1,293 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/message"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type TestCase struct {
+	Identity    *flow.Identity
+	GetIdentity func(pid peer.ID) (*flow.Identity, bool)
+	Channel     channels.Channel
+	Message     interface{}
+	MessageStr  string
+}
+
+func TestIsAuthorizedSender(t *testing.T) {
+	suite.Run(t, new(TestAuthorizedSenderValidatorSuite))
+}
+
+type TestAuthorizedSenderValidatorSuite struct {
+	suite.Suite
+	authorizedSenderTestCases             []TestCase
+	unauthorizedSenderTestCases           []TestCase
+	unauthorizedMessageOnChannelTestCases []TestCase
+}
+
+func (s *TestAuthorizedSenderValidatorSuite) SetupTest() {
+	s.initializeAuthorizationTestCases()
+	s.initializeInvalidMessageOnChannelTestCases()
+}
+
+// TestValidatorCallback_AuthorizedSender checks that the call back returned from AuthorizedSenderValidator does not return false positive
+// validation errors for all possible valid combinations (authorized sender role, message type).
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_AuthorizedSender() {
+	for _, c := range s.authorizedSenderTestCases {
+		str := fmt.Sprintf("role (%s) should be authorized to send message type (%s) on channel (%s)", c.Identity.Role, c.MessageStr, c.Channel)
+		s.Run(str, func() {
+			validate := AuthorizedSenderValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+
+			pid, err := unittest.PeerIDFromFlowID(c.Identity)
+			s.Require().NoError(err)
+
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
nit: You can just create a single context for the whole test method instead of creating it in the loop

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/9086a7e0ae9f81c9a54d847214633fc7b21bb31b